### PR TITLE
Disable known failing device tests using xharness

### PR DESCRIPTION
**Pull request by chamons**
So far these four are disabled:

- https://github.com/xamarin/maccore/issues/1008
- https://github.com/xamarin/maccore/issues/1009
- https://github.com/xamarin/maccore/issues/1014
- https://github.com/xamarin/maccore/issues/1011

I tested locally that TestsToSkipOnDevice.Contains hits but I did not do full device test run locally.

### CODE REVIEW

**Comment by rolfbjarne** [comment]
Issues related to mono should ideally be narrowed down and filed in the mono repo, otherwise they'll never be fixed (even less if we ignore them).

So I had a look again at https://github.com/xamarin/maccore/issues/1014, and luckily I was able to reproduce it, so I fixed it (#4897).

I also tried reproducing the mini timeouts (https://github.com/xamarin/maccore/issues/1011) again, but no luck there. Have you tried it locally?

I'm fine with disabling the F# extension tests, those are useless right now until the corresponding bug is fixed.

The general case of all extension tests is a bit more complicated though, because they do actually confirm one thing even if they crash: that they build. We've almost released XI versions with extension build bugs... and the extension tests were disabled at the time. What's needed here is a way to be able to make tests build-only somehow (https://github.com/xamarin/maccore/issues/1076), which is a somewhat larger project.

**Comment by chamons** [comment]
I can not reproduce #1011 locally. I'll look at making extensions build only and remove #1014 removal.

**Comment by chamons** [comment]
@rolfbjarne I'm going to try to isolate the test failures more, but thought on the PR now?

File: tests/xharness/Jenkins.cs [hunk 1]
```diff
@@ -481,6 +495,20 @@ void SelectTests ()
 			}
 		}
 
+		void DisableKnownFailingDeviceTests ()
+		{
+			// https://github.com/xamarin/maccore/issues/1008
+			ForceExtensionBuildOnly = true;
+
+			// https://github.com/xamarin/maccore/issues/1009 
+			foreach (var fsharp in Harness.IOSTestProjects.Where (x => x.Name == "fsharp" || x.Name == "fsharplibrary"))
```
Comment (rolfbjarne) [inline tests/xharness/Jenkins.cs]: The F# extensions tests don't even build, so these can be completely ignored.

File: tests/xharness/Jenkins.cs [hunk 2]
```diff
@@ -3140,6 +3169,9 @@ public async Task<bool> BuildAsync ()
 
 		protected override async Task ExecuteAsync ()
 		{
+			if (BuildOnly)
```
Comment (rolfbjarne) [inline tests/xharness/Jenkins.cs]: How do these tests show up the html report now (after running them)?

**Pull request merged**